packages/mobile/android/app/src/main/java/com/zbaymobile/MainActivity.java

```diff
@@ -1,15 +1,66 @@
 package com.zbaymobile;
 
 import com.facebook.react.ReactActivity;
+import com.facebook.react.ReactInstanceManager;
+import com.facebook.react.bridge.ReactContext;
+import android.content.Intent;
+import android.content.Context;
+import android.util.AttributeSet;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import com.facebook.react.modules.core.DeviceEventManagerModule;
 
-public class MainActivity extends ReactActivity {
+import com.facebook.react.bridge.ReactApplicationContext;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+public class MainActivity extends ReactActivity  {
 
   /**
    * Returns the name of the main component registered from JavaScript. This is used to schedule
    * rendering of the component.
    */
+
+   private void sendEvent() {
+    Intent intent = getIntent();
+
+    ReactInstanceManager mReactInstanceManager = getReactNativeHost().getReactInstanceManager();
+    ReactApplicationContext reactContext = (ReactApplicationContext) mReactInstanceManager.getCurrentReactContext();
+
+
+    Bundle bundle = intent.getExtras();
+    if (bundle != null) {
+        for (String key : bundle.keySet()) {
```

Comment:
Why do you fire an event for each item instead of serializing extras and triggering single event? 
I think such method should have intent data passed as an argument instead of trying to access it directly.
Also context is something that can be passed as an argument - thanks to that the method can become static which keeps code clean and makes it  easier to test